### ISSUE TITLE
SLF4J too verbose

### ISSUE THREAD

**Issue by DavidPerezIngeniero** [earldouglas/xsbt-web-plugin#224]
I would like to see only ERROR or WARN messages when using `container:start`

I see a lot of output of INFO messages from SLF4J.

I control the logging of my own webapp, but not the messages coming from `Jetty` or `xsbt-web-plugin`.

Here is the output I receive, that isn't interesting for me:

        $ sbt myproject/container:start

        2015-02-03 11:32:24.463:INFO::main: Logging initialized @47ms
	2015-02-03 11:32:24.470:INFO:oejr.Runner:main: Runner
	2015-02-03 11:32:24.555:INFO:oejs.Server:main: jetty-9.2.1.v20140609
	2015-02-03 11:32:33.489:WARN:oeja.AnnotationConfiguration:main: ServletContainerInitializers: detected. Class hierarchy: empty
	2015-02-03 11:32:33.926:INFO:/:main: Starting spray application ...
	2015-02-03 11:32:34.237:INFO:/:main: Automatically setting spray.servlet.root-path to ''
	11:32:34,648 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
	11:32:34,648 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [jar:file:/home/d/rds/servision/target/webapp/WEB-INF/lib/servision_2.11-0.1-SNAPSHOT.jar!/logback-test.xml]
	11:32:34,681 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@4c12d715 - URL [jar:file:/home/d/rds/servision/target/webapp/WEB-INF/lib/servision_2.11-0.1-SNAPSHOT.jar!/logback-test.xml] is not of type file
	11:32:35,066 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
	11:32:35,081 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@806ea2f - Adding [file:/home/d/rds/servision/target/webapp/WEB-INF/classes/logback-base.xml] to configuration watch list.
	11:32:35,101 |-INFO in ch.qos.logback.classic.joran.action.LoggerContextListenerAction - Adding LoggerContextListener of type [ch.qos.logback.classic.jul.LevelChangePropagator] to the object stack
	11:32:35,126 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating DEBUG level on Logger[ROOT] onto the JUL framework
	11:32:35,126 |-INFO in ch.qos.logback.classic.joran.action.LoggerContextListenerAction - Starting LoggerContextListener
	11:32:35,126 |-INFO in ch.qos.logback.classic.joran.action.JMXConfiguratorAction - begin
	11:32:35,451 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word exc with class [ch.qos.logback.classic.pattern.ConversorPilaLlamadas]
	11:32:35,451 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word msj with class [fcc.ima.trazado.LimpiadorMensajes]
	11:32:35,451 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.rolling.RollingFileAppender]
	11:32:35,455 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [Fichero]
	11:32:35,474 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
	11:32:35,526 |-INFO in c.q.l.core.rolling.TimeBasedRollingPolicy - Will use gz compression
	11:32:35,528 |-INFO in c.q.l.core.rolling.TimeBasedRollingPolicy - Will use the pattern ../logs/ima%d{ddMMMyyyy} for the active file
	11:32:35,532 |-INFO in c.q.l.core.rolling.DefaultTimeBasedFileNamingAndTriggeringPolicy - The date pattern is 'ddMMMyyyy' from file name pattern '../logs/ima%d{ddMMMyyyy}.gz'.
	11:32:35,532 |-INFO in c.q.l.core.rolling.DefaultTimeBasedFileNamingAndTriggeringPolicy - Roll-over at midnight.
	11:32:35,539 |-INFO in c.q.l.core.rolling.DefaultTimeBasedFileNamingAndTriggeringPolicy - Setting initial period to Tue Feb 03 11:32:20 CET 2015
	11:32:35,540 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[Fichero] - Active log file name: ../logs/ima.log
	11:32:35,540 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[Fichero] - File property is set to [../logs/ima.log]
	11:32:35,541 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
	11:32:35,543 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [ConsolaAnsi]
	11:32:35,547 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
	11:32:35,551 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
	11:32:35,551 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [Consola]
	11:32:35,551 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
	11:32:35,552 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.classic.net.SMTPAppender]
	11:32:35,563 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [ErrorConversionPorCorreo]
	11:32:35,981 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[fcc] onto the JUL framework
	11:32:35,981 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - fcc level set to INFO
	11:32:35,981 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[es.fcc] onto the JUL framework
	11:32:35,981 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - es.fcc level set to INFO
	11:32:35,981 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[jython] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jython level set to INFO
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[jima] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jima level set to INFO
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating WARN level on Logger[jima.administracion.tareas.JobStore] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jima.administracion.tareas.JobStore level set to WARN
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating WARN level on Logger[jima.administracion.tareas.TareaRima] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jima.administracion.tareas.TareaRima level set to WARN
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to WARN
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating WARN level on Logger[ROOT] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [ConsolaAnsi] to Logger[ROOT]
	11:32:35,984 |-ERROR in ch.qos.logback.core.joran.action.AppenderRefAction - Could not find an appender named [Respuesta]. Did you define it below instead of above in the configuration file?
	11:32:35,984 |-ERROR in ch.qos.logback.core.joran.action.AppenderRefAction - See http://logback.qos.ch/codes.html#appender_order for more details.
	11:32:35,984 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
	11:32:35,986 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@78ded0bb - Registering current configuration as safe fallback point

	SLF4J: The following set of substitute loggers may have been accessed
	SLF4J: during the initialization phase. Logging calls during this
	SLF4J: phase were not honored. However, subsequent logging calls to these
	SLF4J: loggers will work as normally expected.
	SLF4J: See also http://www.slf4j.org/codes.html#substituteLogger
	SLF4J: jima.entorno.area.informe.Birt
	SLF4J: fcc.vision.modelo.Escenarios
	SLF4J: fcc.ima.bd.ChequeadorBaseDatos
	SLF4J: fcc.ima.jython.Jython
	SLF4J: fcc.vision.modelo.Portales
	SLF4J: jima.entorno.area.oda.DatosRds
	SLF4J: fcc.vision.rest.ActorVision
	SLF4J: fcc.ima.web.InicializadorEntorno$
	 2015-02-03 11:32:24.463:INFO::main: Logging initialized @47ms
	2015-02-03 11:32:24.470:INFO:oejr.Runner:main: Runner
	2015-02-03 11:32:24.555:INFO:oejs.Server:main: jetty-9.2.1.v20140609
	2015-02-03 11:32:33.489:WARN:oeja.AnnotationConfiguration:main: ServletContainerInitializers: detected. Class hierarchy: empty
	2015-02-03 11:32:33.926:INFO:/:main: Starting spray application ...
	2015-02-03 11:32:34.237:INFO:/:main: Automatically setting spray.servlet.root-path to ''
	11:32:34,648 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
	11:32:34,648 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [jar:file:/home/d/rds/servision/target/webapp/WEB-INF/lib/servision_2.11-0.1-SNAPSHOT.jar!/logback-test.xml]
	11:32:34,681 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@4c12d715 - URL [jar:file:/home/d/rds/servision/target/webapp/WEB-INF/lib/servision_2.11-0.1-SNAPSHOT.jar!/logback-test.xml] is not of type file
	11:32:35,066 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
	11:32:35,081 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@806ea2f - Adding [file:/home/d/rds/servision/target/webapp/WEB-INF/classes/logback-base.xml] to configuration watch list.
	11:32:35,101 |-INFO in ch.qos.logback.classic.joran.action.LoggerContextListenerAction - Adding LoggerContextListener of type [ch.qos.logback.classic.jul.LevelChangePropagator] to the object stack
	11:32:35,126 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating DEBUG level on Logger[ROOT] onto the JUL framework
	11:32:35,126 |-INFO in ch.qos.logback.classic.joran.action.LoggerContextListenerAction - Starting LoggerContextListener
	11:32:35,126 |-INFO in ch.qos.logback.classic.joran.action.JMXConfiguratorAction - begin
	11:32:35,451 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word exc with class [ch.qos.logback.classic.pattern.ConversorPilaLlamadas]
	11:32:35,451 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word msj with class [fcc.ima.trazado.LimpiadorMensajes]
	11:32:35,451 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.rolling.RollingFileAppender]
	11:32:35,455 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [Fichero]
	11:32:35,474 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
	11:32:35,526 |-INFO in c.q.l.core.rolling.TimeBasedRollingPolicy - Will use gz compression
	11:32:35,528 |-INFO in c.q.l.core.rolling.TimeBasedRollingPolicy - Will use the pattern ../logs/ima%d{ddMMMyyyy} for the active file
	11:32:35,532 |-INFO in c.q.l.core.rolling.DefaultTimeBasedFileNamingAndTriggeringPolicy - The date pattern is 'ddMMMyyyy' from file name pattern '../logs/ima%d{ddMMMyyyy}.gz'.
	11:32:35,532 |-INFO in c.q.l.core.rolling.DefaultTimeBasedFileNamingAndTriggeringPolicy - Roll-over at midnight.
	11:32:35,539 |-INFO in c.q.l.core.rolling.DefaultTimeBasedFileNamingAndTriggeringPolicy - Setting initial period to Tue Feb 03 11:32:20 CET 2015
	11:32:35,540 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[Fichero] - Active log file name: ../logs/ima.log
	11:32:35,540 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[Fichero] - File property is set to [../logs/ima.log]
	11:32:35,541 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
	11:32:35,543 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [ConsolaAnsi]
	11:32:35,547 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
	11:32:35,551 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
	11:32:35,551 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [Consola]
	11:32:35,551 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
	11:32:35,552 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.classic.net.SMTPAppender]
	11:32:35,563 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [ErrorConversionPorCorreo]
	11:32:35,981 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[fcc] onto the JUL framework
	11:32:35,981 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - fcc level set to INFO
	11:32:35,981 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[es.fcc] onto the JUL framework
	11:32:35,981 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - es.fcc level set to INFO
	11:32:35,981 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[jython] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jython level set to INFO
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating INFO level on Logger[jima] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jima level set to INFO
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating WARN level on Logger[jima.administracion.tareas.JobStore] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jima.administracion.tareas.JobStore level set to WARN
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating WARN level on Logger[jima.administracion.tareas.TareaRima] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - jima.administracion.tareas.TareaRima level set to WARN
	11:32:35,982 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to WARN
	11:32:35,982 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@7df8b7c6 - Propagating WARN level on Logger[ROOT] onto the JUL framework
	11:32:35,982 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [ConsolaAnsi] to Logger[ROOT]
	11:32:35,984 |-ERROR in ch.qos.logback.core.joran.action.AppenderRefAction - Could not find an appender named [Respuesta]. Did you define it below instead of above in the configuration file?
	11:32:35,984 |-ERROR in ch.qos.logback.core.joran.action.AppenderRefAction - See http://logback.qos.ch/codes.html#appender_order for more details.
	11:32:35,984 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
	11:32:35,986 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@78ded0bb - Registering current configuration as safe fallback point

	SLF4J: The following set of substitute loggers may have been accessed
	SLF4J: during the initialization phase. Logging calls during this
	SLF4J: phase were not honored. However, subsequent logging calls to these
	SLF4J: loggers will work as normally expected.
	SLF4J: See also http://www.slf4j.org/codes.html#substituteLogger
	SLF4J: jima.entorno.area.informe.Birt
	SLF4J: fcc.vision.modelo.Escenarios
	SLF4J: fcc.ima.bd.ChequeadorBaseDatos
	SLF4J: fcc.ima.jython.Jython
	SLF4J: fcc.vision.modelo.Portales
	SLF4J: jima.entorno.area.oda.DatosRds
	SLF4J: fcc.vision.rest.ActorVision
	SLF4J: fcc.ima.web.InicializadorEntorno$

How can I remove this noise?

**Comment by earldouglas** [comment]
Wow, that's noisy.  Are you certain it's not coming from your application configuration?  What level is your root logger set to?

**Comment by DavidPerezIngeniero** [comment]
My logger is set to INFO, but the traces are colourful, and I can tell
easily the difference between the ones that come from my webapp, that the
ones that come from Jetty.

2015-02-04 3:49 GMT+01:00 James Earl Douglas <notifications@github.com>:

**Comment by earldouglas** [comment]
I think this should be controllable via `outputStrategy` in `ForkOptions`, but I'd like to do a bit more investigation.

Any chance you can provide a sample configuration that reproduces this behavior?

**Comment by earldouglas** [comment]
The quick and dirty approach would be to use an sbt `Logger` to capture Jetty's StdErrLog output:

*build.sbt:*

```scala
jetty(options = new ForkOptions(outputStrategy = Some(LoggedOutput(new ConsoleLogger))))
```

See:
* http://www.scala-sbt.org/0.13.7/api/index.html#sbt.LoggedOutput
* http://www.scala-sbt.org/0.13.7/api/index.html#sbt.ConsoleLogger$

A more robust approach would be to configure Jetty's logging to use something more intentional:

* http://www.eclipse.org/jetty/documentation/current/configuring-logging.html#configuring-jetty-logging

*build.sbt:*

```scala
jetty(config = "etc/jetty.xml")
```

*etc/jetty.xml:*

```xml
<Configure id="Server" class="org.eclipse.jetty.server.Server">
  ...
  <Call class="org.eclipse.jetty.util.log.Log" name="getRootLogger">
    <Call name="setLevel">
      <Arg type="int">10</Arg>
    </Call>
  </Call>
</Configure>
```

**Comment by earldouglas** [comment]
Closing this one, but please let me know if you aren't able to get the above configuration working.

**Comment by tek** [comment]
could you clarify the second example in the doc? I'm trying to get jetty to use logback, but it stubbornly refuses to do so. I tried
```xml
  <Set class="org.eclipse.jetty.util.log.Log" name="rootLogger">
    <New class="org.eclipse.jetty.util.log.Slf4jLog"/>
    <New class="ch.qos.logback.classic.Logger"/>
  </Set>
```
(the two classes alternatively), but both fail with `ClassNotFound` for `org.slf4j.Logger` for the former and the actual class for the latter. Both are present in `WEB-INF/lib` and logback is in use for everything except jetty output. I guess if jetty would see them, it would use them automatically anyway, so it seems that the problem is that jetty can't access the whole classpath at that point…any ideas?

**Comment by earldouglas** [comment]
Try adding the `jetty-webapp-logging` library to your container:

```scala
containerLibs in Container += "org.eclipse.jetty" %  "jetty-webapp-logging" % "9.1.0.v20131115"
```

**Comment by tek** [comment]
I added webapp-logging as a regular dep before, but even using `containerLibs` and adding logback and slf4j, it doesn't make a difference, very weird.

**Comment by earldouglas** [comment]
I'll try it out later tonight and get back to you when I figure it out.

**Comment by earldouglas** [comment]
Looks like you need something more like this:

```scala
containerLibs in Jetty ++= Seq(
  "org.eclipse.jetty" %  "jetty-webapp-logging" % "9.0.0",
  "org.slf4j" % "slf4j-simple" % "1.7.12"
)
```

Though the *jetty.xml* configuration doesn't look quite right on my instance.

**Comment by tek** [comment]
ah, of course :D it works with this `jetty.xml`:
```xml
<Configure id="server" class="org.eclipse.jetty.server.Server">
  <Set class="org.eclipse.jetty.util.log.Log" name="log">
    <New class="org.eclipse.jetty.util.log.Slf4jLog"/>
  </Set>
</Configure>
```
Problem is though, `logback.xml` is being ignored, so jetty spouts thousands of lines of debug output. I guess that maybe `${jetty.home}` isn't set to the sbt `baseDirectory` so the `resources` dir isn't visible to jetty.

**Comment by tek** [comment]
Alright, I got it working.
First of all, the `jetty.xml` is unnecessary when the logging jars are on the cp, it defaults to slf4j if it is available.
Secondly, I think the resources dir has to be enabled like [here](https://github.com/eclipse/jetty.project/blob/master/jetty-server/src/main/config/modules/resources.mod) in order to be automatically read by jetty, but no idea how this would work with *jetty-runner*.
**Anyhoo**, I got it working quite simply, by bruteforcing the logback config like so:
```scala
javaOptions in Jetty += s"-Dlogback.configurationFile=${(resourceDirectory in Compile).value}/logback.xml"
```
I guess there's probably a better way.
